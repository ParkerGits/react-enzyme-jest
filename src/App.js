import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>

const Title = ({text}) => <div>{text}</div>

class App extends Component {
  state = {
     on: false,
     input: '',
     mainColor: 'blue',
     lifeCycle: ''
  }
  componentDidMount() {
    this.setState({lifeCycle: 'componentDidMount'})
  }
  componentWillReceiveProps() {
    this.setState({lifeCycle: 'componentWillReceiveProps'})
  }
  render() {
      return(
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1>Welcome to React</h1>
          <h3 className={this.state.mainColor}>Everyone is Welcome!</h3>
          <ul className="tyler">
            <li>Test 1</li>
            <li>Test 2</li>
            <li>Test 3</li>
          </ul>
          <li>Test 3</li>
          <p className='lifeCycle'>{this.state.lifeCycle}</p>
          <p className='button-state'>{this.state.on ? 'Yes!' : 'No!'}</p>
          <button onClick = {() => this.setState({on: true})}>Click</button>
          <h2>{this.state.input}</h2>
          <input onChange={(e) => this.setState({input: e.currentTarget.value})} type='text'/>
          <p className="App-intro">
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <Title text="Some title" />
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          <p>Hello World</p>
        </header>
        <Test />
      </div>
    );
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>
  }
}

export default App;
