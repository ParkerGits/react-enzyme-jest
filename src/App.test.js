import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('<App />', () => {
  const wrapper = shallow(<App />)
  it('should contain an element with logo as alt tag', () => {
    expect(wrapper.find({alt: 'logo'}).text()).toBe('Welcome to React')
  })
})
