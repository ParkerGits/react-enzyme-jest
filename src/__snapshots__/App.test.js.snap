// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1>
        Welcome to React
      </h1>
      <h3
        className="blue"
      >
        Everyone is Welcome!
      </h3>
      <ul
        className="tyler"
      >
        <li>
          Test 1
        </li>
        <li>
          Test 2
        </li>
        <li>
          Test 3
        </li>
      </ul>
      <li>
        Test 3
      </li>
      <p
        className="button-state"
      >
        No!
      </p>
      <button
        onClick={[Function]}
      >
        Click
      </button>
      <h2 />
      <input
        onChange={[Function]}
        type="text"
      />
      <p
        className="App-intro"
      >
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <Title
        text="Some title"
      >
        <div>
          Some title
        </div>
      </Title>
      <a
        className="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <p>
        Hello World
      </p>
    </header>
    <Test>
      <div>
        Testing
      </div>
    </Test>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1>
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is Welcome!
    </h3>
    <ul
      className="tyler"
    >
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
      <li>
        Test 3
      </li>
    </ul>
    <li>
      Test 3
    </li>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <p
      className="App-intro"
    >
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Title
      text="Some title"
    />
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <p>
      Hello World
    </p>
  </header>
  <Test />
</div>
`;
